File: fixtures/ts-basic/src/repositories/orderRepository.ts
import { Order } from "../models/order";

export class OrderRepository {
  private orders = new Map<string, Order>();

  findById(id: string): Order | undefined {
    return this.orders.get(id);
  }

  findByUserId(userId: string): Order[] {
    return Array.from(this.orders.values()).filter((o) => o.userId === userId);
  }

  insert(order: Order): void {
    this.orders.set(order.id, order);
  }

  updateStatus(id: string, status: Order["status"]): Order | undefined {
    const order = this.orders.get(id);
    if (!order) return undefined;
    order.status = status;
    return order;
  }
}

File: fixtures/csharp-basic/Repositories/OrderRepository.cs
using System.Collections.Generic;
using OrderSystem.Models;

namespace OrderSystem.Repositories;

public class OrderRepository : IOrderRepository
{
    private readonly List<Order> _orders = new List<Order>();

    public void Save(Order order)
    {
        _orders.Add(order);
    }

    public Order FindById(int id)
    {
        foreach (var order in _orders)
        {
            if (order.Id == id)
            {
                return order;
            }
        }
        return null;
    }
}
